Switch license to mixed CC BY 4.0 and Apache 2.0

keryell opened the pull request:
This is required to have both document content and both infrastructure
at the same time.
Align on OpenCL licensing schema.

oddhack commented:
@keryell is this approved to merge and you're asking me to do it? Not sure why I'm assigned.

keryell commented:
> @keryell is this approved to merge and you're asking me to do it? Not sure why I'm assigned.

I just want some feedback about the minimal licensing effort. It looks like each `-Docs` repository has its own variant and it is not clear how to phrase it...

oddhack commented:
This looks fine to me, if the WG is OK with it. Let me know when you want it merged, unless you want to do that yourself.

oddhack commented:
> This makes sense as the top level change, but do we also need to add info about the covering license to the individual files? This change suggests that the individual files already have the information, but I'm not sure that they do

I'm not entirely sure if this is what you're asking, but each file should have exactly one of CC-BY or Apache; it's not the files that are being dual-licensed, but the repository.

mkinsner commented:
> The files in, and generated output documents from this SYCL-Docs project are
under a mix of copyright and license statements. Refer to the individual files
for specific information.

I'm asking whether we need to clarify in each file which license it's covered by, once we enable a mixed license for the repo.  For example, does each *.tex file need a header saying which license it is under?

keryell commented:
For me it is implicitly (and lazily) covered by the sentence "As a general guideline:" from `LICENSE.txt`.

The C++ headers and examples are directly included in LaTeX the specification so they cannot have license header or it would be displayed in the specification...

oddhack commented:
Yes, each file needs a copyright statement. They aren't long. Strongly suggest not trying to invent new copyright law on the fly :-) Confirm with Neil if you're unsure about this.

oddhack commented:
@keryell the copyright statement on the file goes in comments, they aren't processed by LaTeX.

keryell commented:
> @keryell the copyright statement on the file goes in comments, they aren't processed by LaTeX.

It is not how the SYCL specification is written.
The headers and code samples are *real* C++ code, not some artistic representation using some mark-up language.
Feel free to rewrite it during your spare time.

oddhack commented:
This isn't a good conversation to have in public. I opened an internal issue on the gitlab repo and have asked Neil to participate.

fraggamuffin commented:
Aug 12:
*  Waiting John L, and send to Neil

fraggamuffin commented:
* aim for first merge of the 2 license files soon, John may need a header or copyright in every file.

keryell commented:
On-going internal discussion at https://gitlab.khronos.org/sycl/Specification/issues/257

keryell commented:
I have added the requested license comments on all the files but the code samples & headers.
I have added a general comment for them in the top-level license file.

oddhack commented:
I didn't look at all 91 files but it looks OK to me.

keryell commented:
> I didn't look at all 91 files but it looks OK to me.

Thanks.
But this is still a work-in-progress because now I need to add the code to skip the display of 91 licenses in the specification... :-/

keryell commented:
> But this is still a work-in-progress because now I need to add the code to skip the display of 91 licenses in the specification... :-/

I think it is ready for prime time now.

I have added the LaTeX code to skip the display of the license texts in the specification and fixed a few bugs around.

oddhack commented on LICENSE.txt:
```diff
@@ -1,9 +1,18 @@
+  license.  Details of the license are at:
+  https://creativecommons.org/licenses/by/4.0/
+
+- the source files for building the specification, such as Makefiles
```
Maybe also say "headers" as the .h files are Apache 2.0? Otherwise, this looks good to me. Thanks for working through the unpleasant license/legal issues!

keryell replied:
Done.
Fortunately there was some LaTeX developments required to add a little bit of fun to this administrative chore... :-)
